firmware/application/app_settings.cpp

```diff
@@ -36,6 +38,105 @@ namespace fs = std::filesystem;
 using namespace portapack;
 using namespace std::literals;
 
+namespace {
+fs::path get_settings_path(const std::string& app_name) {
+    return fs::path{u"/SETTINGS"} / app_name + u".ini";
+}
+}  // namespace
+
+std::string Setting::to_string() const {
+    auto result = std::string{name_} + "=";
+
+    if (std::holds_alternative<uint32_t>(value_))
+        result += to_string_dec_uint(as_uint());
```

Comment:
It would be nice if negative numbers and U64's were supported.  The Recon app, for example, makes use of several signed numbers.  And any Frequency value >4GHz needs >32 bits.

Reply:
I think the variant ends up using the length of its largest alternative. Maybe instead of uint32 we just use int64? That would provide for signed numbers and be long enough for everything we want.

Reply:
I say go for int64_t !